vimeo-networking/src/main/java/com/vimeo/networking/utils/ISO8601Wrapper.java

```diff
@@ -0,0 +1,55 @@
+package com.vimeo.networking.utils;
+
+import com.google.gson.JsonDeserializationContext;
+import com.google.gson.JsonDeserializer;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonParseException;
+import com.google.gson.JsonPrimitive;
+import com.google.gson.JsonSerializationContext;
+import com.google.gson.JsonSerializer;
+import com.google.gson.internal.bind.util.ISO8601Utils;
+import com.vimeo.networking.logging.ClientLogger;
+
+import java.lang.reflect.Type;
+import java.text.ParseException;
+import java.text.ParsePosition;
+import java.util.Date;
+
+/**
+ * A wrapper class that serializes and deserializes dates
+ * using the {@link ISO8601Utils} class that catches errors
+ * and reports them to the client application, allowing
+ * them to decide if they should crash or not. If we just
+ * rely on the default adapter used by JSON, we are unable
+ * to absorb date parsing errors or log them correctly.
+ */
+@SuppressWarnings("WeakerAccess")
+public final class ISO8601Wrapper {
+
+    private ISO8601Wrapper() {
+    }
+
+    public static JsonSerializer<Date> getDateSerializer() {
+        return new JsonSerializer<Date>() {
+            @Override
+            public JsonElement serialize(Date src, Type typeOfSrc, JsonSerializationContext context) {
+                return src == null ? null : new JsonPrimitive(ISO8601Utils.format(src));
+            }
+        };
+    }
+
+    public static JsonDeserializer<Date> getDateDeserializer() {
+        return new JsonDeserializer<Date>() {
+            @Override
+            public Date deserialize(JsonElement json, Type typeOfT, JsonDeserializationContext context)
+                    throws JsonParseException {
```

Comment:
It doesn't actually throw a JsonParseException, so maybe ditch the the "throws"? It'll still be a legit overrride.

Also, if you just look at the javadocs for getAsString there are other types of exceptions that can occur here due to getAsString. Maybe we should swallow more than just ParseException? But perhaps more importantly, unless I am missing something...

Here's the weird thing regarding getAsString, in the version of gson that I have, this method always throws an UnsupportedOperationException. I'm confused...